designatedlands.py

```diff
@@ -0,0 +1,1283 @@
+        layer = fiona.listlayers(os.path.join(out_folder, filename))[0]
+    return (out_file, layer)
+
+
+class ZipCompatibleTarFile(tarfile.TarFile):
+    """
+    Wrapper around TarFile to make it more compatible with ZipFile
+    Modified from https://github.com/OpenBounds/Processing/blob/master/utils.py
+    """
+
+    def infolist(self):
+        members = self.getmembers()
+        for m in members:
+            m.filename = m.name
+        return members
+
+    def namelist(self):
+        return self.getnames()
+
+
+def get_compressed_file_wrapper(path):
+    """ From https://github.com/OpenBounds/Processing/blob/master/utils.py
+    """
+    ARCHIVE_FORMAT_ZIP = "zip"
+    ARCHIVE_FORMAT_TAR_GZ = "tar.gz"
+    ARCHIVE_FORMAT_TAR_BZ2 = "tar.bz2"
+    archive_format = None
+    if path.endswith(".zip"):
+        archive_format = ARCHIVE_FORMAT_ZIP
+    elif path.endswith(".tar.gz") or path.endswith(".tgz"):
+        archive_format = ARCHIVE_FORMAT_TAR_GZ
+    elif path.endswith(".tar.bz2"):
+        archive_format = ARCHIVE_FORMAT_TAR_BZ2
+    else:
+        try:
+            with zipfile.ZipFile(path, "r") as f:
+                archive_format = ARCHIVE_FORMAT_ZIP
+        except:
+            try:
+                f = tarfile.TarFile.open(path, "r")
+                f.close()
+                archive_format = ARCHIVE_FORMAT_ZIP
+            except:
+                pass
+    if archive_format is None:
+        raise Exception("Unable to determine archive format")
+
+    if archive_format == ARCHIVE_FORMAT_ZIP:
+        return zipfile.ZipFile(path, "r")
+
+    elif archive_format == ARCHIVE_FORMAT_TAR_GZ:
+        return ZipCompatibleTarFile.open(path, "r:gz")
+
+    elif archive_format == ARCHIVE_FORMAT_TAR_BZ2:
+        return ZipCompatibleTarFile.open(path, "r:bz2")
+
+
+class DesignatedLands(object):
+    """ A class to hold the job's config, data and methods
+    """
+
+    def __init__(self, config_file=None):
+
+        LOG.info("Initializing designatedlands")
+
+        # load default config
+        self.config = DEFAULT_CONFIG.copy()
+
+        # if provided with a config file, replace config values with those present in
+        # thie config file
+        if config_file:
+            if not os.path.exists(config_file):
+                raise ConfigValueError(f"File {config_file} does not exist")
+            self.read_config(config_file)
+
+        # set default n_processes to the number of cores available minus one
+        if self.config["n_processes"] == -1:
+            self.config["n_processes"] = multiprocessing.cpu_count() - 1
+
+        # don't try and use more cores than are available
+        elif self.config["n_processes"] > multiprocessing.cpu_count():
+            self.config["n_processes"] = multiprocessing.cpu_count()
+
+        self.db = pgdata.connect(self.config["db_url"])
+        self.db.ogr_string = f"PG:host={self.db.host} user={self.db.user} dbname={self.db.database} port={self.db.port}"
```

Comment:
I got an error with `gdal_rasterize` saying no password supplied. This fixed it for me:

```suggestion
        self.db.ogr_string = f"PG:host={self.db.host} user={self.db.user} dbname={self.db.database} password={self.db.password} port={self.db.port}"
```